nexus-sdk/src/main/java/io/nexusrpc/OperationHandler.java

```diff
@@ -0,0 +1,228 @@
+package io.nexusrpc;
+
+import java.time.Duration;
+import java.util.Map;
+import java.util.function.BiFunction;
+import org.jspecify.annotations.Nullable;
+
+/**
+ * Handler for an operation.
+ *
+ * <p>An instance of this must be returned from an {@link OperationImpl} annotated method in a
+ * {@link ServiceImpl}. Simple synchronous operations can use {@link #sync}, more complex or
+ * asynchronous operations can manually implement this interface.
+ *
+ * @param <T> The parameter type of the operation. This can be {@link Void} for no parameter.
+ * @param <R> the return type of the operation. This can be {@link Void} for no return.
+ */
+public interface OperationHandler<T, R> {
+  /**
+   * Create an operation handler for a synchronous operation backed by the given function. This
+   * function will be called for each operation invocation. All other calls to the operation are not
+   * supported.
+   */
+  static <T, R> OperationHandler<T, R> sync(BiFunction<StartContext, @Nullable T, R> func) {
+    return new OperationHandler<T, R>() {
+      @Override
+      public StartResult<R> start(StartContext ctx, @Nullable T param) {
+        return StartResult.sync(func.apply(ctx, param));
+      }
+
+      @Override
+      public R fetchResult(FetchResultContext ctx) {
+        throw new UnsupportedOperationException("Not supported on sync operation");
+      }
+
+      @Override
+      public OperationInfo fetchInfo(FetchInfoContext ctx) {
+        throw new UnsupportedOperationException("Not supported on sync operation");
+      }
```

Comment:
Called `GetInfo` and `GetResult` in Go (although I'm considering renaming to `Info` and `Result` since getters in Go typically don't have the `Get` prefix.

Reply:
I think in this case we can live without cross lang consistency.

Reply:
Yes, this is a case where it's more important to match Java than Go